src/data_structures/suffix_array.rs

```diff
@@ -19,12 +20,79 @@ use vec_map::VecMap;
 
 use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
+use data_structures::bwt::{BWT, Less, Occ};
 
+pub trait SuffixArray {
+    fn get(&self, index: usize) -> Option<usize>;
+    fn range(&self, range: Range<usize>) -> Option<Vec<usize>>;
```

Comment:
If an iterator would fit all forseen use cases for `range`, then I agree. I would actually lean more towards making new custom struct that lazily computes .get()'d indices for the SampledSuffixArrray and vice versa. I wasn't sure what `range` might be useful for, so I tried to expose the underlying types as much as possible. Stepping back for a moment-- what is `range` actually for? Is it primarily for testing? Do we even need it, or can users suffice with `get` and `len`?